invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/UppyUploader/InvenioMultipartUploader.js

```diff
@@ -0,0 +1,384 @@
+   *   - **fields**: object, an object with form fields to send along with the upload request. For presigned PUT uploads (which are default), this should be left empty.
+   *   - **headers**: object, an object with request headers to send along with the upload request. When using a presigned PUT upload, it’s a good idea to provide headers['content-type']. That will make sure that the request uses the same content-type that was used to generate the signature. Without it, the browser may decide on a different content-type instead, causing S3 to reject the upload.
+   */
+  async getUploadParameters(file, options, signal) {
+    file.transferOptions = {
+      fileSize: file.size,
+      type: "L",
+      cancelFn: () => this.uppy.removeFile(file.id),
+    };
+
+    try {
+      const response = await this.opts.initializeFileUpload(this.draftRecord, file);
+      this.uppy.setFileMeta(file.id, {
+        file_id: response.file_id,
+        links: response.links,
+      });
+      return {
+        headers: {
+          "Content-Type": "application/octet-stream",
+        },
+        url: response.links.content,
+        method: "PUT",
+        signal,
+      };
+    } catch (error) {
+      this.uppy.info(error.message, "error");
+      throw error;
+    }
+  }
+
+  /**
+   * A function that returns the minimum chunk size to use when uploading the given file as multipart.
+   * For multipart uploads, chunks are sent in batches to have presigned URLs generated with signPart(). To reduce the amount of requests for large files, you can choose a larger chunk size, at the cost of having to re-upload more data if one chunk fails to upload.
+   * S3 requires a minimum chunk size of 5MiB, and supports at most 10,000 chunks per multipart upload. If getChunkSize() returns a size that’s too small, Uppy will increase it to S3’s minimum requirements.
+   * Default implementation produces a chunk size between 5-250 MB for max. 10240 chunks (max filesize. ~2.44 TiB).
+   * @param {*} file the file object from Uppy’s state. The most relevant keys are file.name and file.type
+   * @returns byte size of individual part chunk
+   */
+  getChunkSize(file) {
+    const MiB = 1024 * 1024;
+    const minPartSize = MiB * 5;
+    const midPartSize = MiB * 25;
+    const maxPartSize = MiB * 250;
+
+    const smallFile = file.size <= this.#maxMultipartParts * minPartSize;
+    const mediumFile = file.size <= this.#maxMultipartParts * midPartSize;
+    const largeFile = file.size <= this.#maxMultipartParts * maxPartSize;
+
+    const chunkSize = smallFile
+      ? minPartSize
+      : mediumFile
+      ? midPartSize
+      : largeFile
+      ? maxPartSize
+      : undefined;
+
+    if (chunkSize === undefined) {
```

Comment:
```suggestion
    if (chunkSize === maxChunkSize) {
```
I find it a bit confusing to compare with undefined